core/src/avm2/globals/array.rs

```diff
@@ -309,6 +327,36 @@ impl<'gc> ArrayIter<'gc> {
             None
         }
     }
+
+    /// Get the next item from the back of the array.
+    ///
+    /// Since this isn't a real iterator, this comes pre-enumerated; it yields
+    /// a pair of the index and then the value.
+    pub fn next_back(
+        &mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+    ) -> Option<Result<(u32, Value<'gc>), Error>> {
+        if self.index < self.rev_index {
+            self.rev_index -= 1;
+
+            let i = self.rev_index;
+
+            Some(
+                self.array_object
+                    .get_property(
```

Comment:
Not necessarily related to this PR, but I think we should have a `get_element` method that receives `u32` (like we have in AVM1).